packages/go_router/lib/src/path_parser.dart

```diff
@@ -0,0 +1,69 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+final _parameterRegExp = RegExp(r':(\w+)(\((?:\\.|[^\\()])+\))?');
+
+/// Creates a [RegExp] that matches a [pattern] specification.
```

Comment:
Could you include an explanation somewhere in these comments an explanation of what a "pattern specification" is? I have no idea what this method does based on reading this comment, since "pattern" is a very broad term.